pkg/kubelet/cm/memorymanager/policy_static.go

```diff
@@ -45,6 +45,8 @@ type staticPolicy struct {
 	systemReserved systemReservedMemory
 	// topology manager reference to get container Topology affinity
 	affinity topologymanager.Store
+	// init container's memory and hugepages reservation that can be reused by app containers in the pod
+	memoryToReuse map[string]map[v1.ResourceName]state.Block
```

Comment:
> When we are talking about the topology manager container policy, it possible that memory allocation for containers inside of the same pod will be done from different NUMA nodes, and I do not see any checks related to where the block was allocated. Please point me to the relevant checks in case if I missed it.
> 
> For example, I can have two init containers, the memory manager allocated 250Mi of the memory for the first init container from the NUMA node 0, and for the second init container allocated 500Mi from the NUMA node 1. So the amount of re-usable memory for the pod is different for each NUMA node.

IIUC, right now, InitContainer2 must Inherit resources, thus the same NUMA node, as InitContainer1. So there won't be a situation, when you have two InitContainers on different NUMA nodes, regardless Topology Manager policy.

Reply:
According to your example and implementation in this PR:
- second init container will inherit 250Mi from first init container and will allocate additional 250Mi from the same NUMA node
- it can't be placed entirely on different NUMA node